internal/stanza/receiver.go

```diff
@@ -113,21 +115,23 @@ func (r *receiver) consumerLoop(ctx context.Context) {
 	// Don't create done channel on every iteration.
 	doneChan := ctx.Done()
 	pLogsChan := r.converter.OutChannel()
-
 	for {
 		select {
 		case <-doneChan:
 			r.logger.Debug("Consumer loop stopped")
 			return
 
 		case pLogs, ok := <-pLogsChan:
+			obsrecvCtx := r.obsrecv.StartLogsOp(ctx)
 			if !ok {
 				r.logger.Debug("Converter channel got closed")
 				continue
```

Comment:
Should we always expect that EndLogsOp is called for each StartLogsOp? In this path this would not be the case and I am not sure if that's ok or not. I looked at EndsLogsOp and I think that the only meaningful result of that would be not ending the observability span explicitly, so not sure if it hurts (that would be rather minor) or not but thought it might be worth calling out

Reply:
Good point. I agree that `obsrecvCtx := r.obsrecv.StartLogsOp(ctx)` should be placed right before `cErr := r.consumer.ConsumeLogs(ctx, pLogs)`

Reply:
Thanks!